fluxcd-kustomize-mutating-webhook.yaml

```
@@ -0,0 +1,74 @@
+package:
+  name: fluxcd-kustomize-mutating-webhook
+  version: "0.5.0"
+  epoch: 0
+  description: A dynamic solution to patch FluxCD Kustomization resources, seamlessly integrating and federating substitution variables across multiple namespaces.
+  copyright:
+    - license: Apache-2.0
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/xunholy/fluxcd-kustomize-mutating-webhook
+      tag: kustomize-mutating-webhook-${{package.version}}
+      expected-commit: 694bac9d7eb73d827147ec0cb22c51cf4a31d5fe
+
+  - uses: go/bump
+    with:
+      deps: |-
+        github.com/go-chi/chi/v5@v5.2.2
+        golang.org/x/net@v0.38.0
+
+  - uses: go/build
+    with:
+      packages: ./cmd/webhook/
+      output: webhook
+
+subpackages:
+  - name: ${{package.name}}-compat
+    description: "Compatibility package to place binary in the location expected by fluxcd-kustomize-mutating-webhook"
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}
+          ln -sf /usr/bin/webhook ${{targets.contextdir}}/webhook
+    test:
+      pipeline:
+        - runs: |
+            test "$(readlink /webhook)" = "/usr/bin/webhook"
+
+update:
+  enabled: true
+  github:
+    identifier: xunholy/fluxcd-kustomize-mutating-webhook
+    strip-prefix: kustomize-mutating-webhook-
+
+test:
+  environment:
+    contents:
+      packages:
+        - openssl
+        - curl
+  pipeline:
+    - uses: test/kwok/cluster
+    - name: Test operator
+      uses: test/daemon-check-output
+      with:
+        setup: |
+          mkdir -p /etc/config /etc/webhook/certs
+
+          cat <<EOF > /etc/config/app-vars.yaml
+          MY_GLOBAL_FOO: "foo-value"
+          ANOTHER_SETTING: "bar"
+          EOF
+
+          openssl req -newkey rsa:2048 -nodes -keyout /etc/webhook/certs/tls.key \
+            -x509 -days 365 -out /etc/webhook/certs/tls.crt \
+            -subj "/CN=localhost"
+        start: |
+          webhook
+        timeout: 10
+        expected_output: |
+          Starting the webhook server on :8443
+        post: |
+          # we use -k flag here to skip the certificate verification, else curl verification fails since cert is self signed
+          curl -k https://localhost:8443/ready
```

```suggestion
          curl --cacert /etc/webhook/certs/tls.crt https://localhost:8443/ready
```
We could use the self-signed cert like this